### PR TITLE
nfc: Fix handling 0 size data in nfc_platform

### PR DESCRIPTION
**Pull request by dchat-nordic**
Fix handling data with 0 size when issuing a callback.

Jira: NCSDK-25211

### CODE REVIEW

File: doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst [hunk 1]
```diff
@@ -650,7 +650,7 @@ Libraries for networking
 Libraries for NFC
 -----------------
 
-|no_changes_yet_note|
+* Fixed an issue with handling 0 size data in :file:`platform_internal_thread` file.
```
Comment (peknis) [inline doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst:653]:
```suggestion
* Fixed an issue with handling zero size data in the :file:`platform_internal_thread` file.
```

**Review by peknis — APPROVED** [review]
Approving with minor nits.

File: subsys/nfc/lib/platform_internal_thread.c [hunk 2]
```diff
@@ -24,8 +24,12 @@ LOG_MODULE_DECLARE(nfc_platform, CONFIG_NFC_PLATFORM_LOG_LEVEL);
 
 #define NFC_LIB_CTX_SIZE CONFIG_NFC_LIB_CTX_MAX_SIZE
 
+#define NFC_HDR_FLAG_NONE 0
+#define NFC_HDR_FLAG_COPY 1
```
Comment (maje-emb) [inline subsys/nfc/lib/platform_internal_thread.c:28]: Consider using the BIT macro.
```suggestion
#define NFC_HDR_FLAG_NONE BIT(0)
#define NFC_HDR_FLAG_COPY BIT(1)
```

File: subsys/nfc/lib/platform_internal_thread.c [hunk 3]
```diff
@@ -24,8 +24,12 @@ LOG_MODULE_DECLARE(nfc_platform, CONFIG_NFC_PLATFORM_LOG_LEVEL);
...
+
 struct nfc_item_header {
-	uint16_t ctx_size;
+	uint8_t ctx_size;
```
Comment (maje-emb) [inline subsys/nfc/lib/platform_internal_thread.c:31]: Shouldn't we add a range for `config NFC_LIB_CTX_MAX_SIZE`?

Comment (dchat-nordic) [reply subsys/nfc/lib/platform_internal_thread.c:30]: Added

**Review by maje-emb — APPROVED** [review]

**Review by KAGA164 — APPROVED** [review]
LGTM

**Review by grochu — APPROVED** [review]

**Pull request merged**